src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1140,17 +1140,29 @@ public ResultSet getProcedureColumns(
       String procedureNameUnparsed = resultSetStepOne.getString("arguments").trim();
       String procedureNameNoArgs = resultSetStepOne.getString("name");
       String schemaName = resultSetStepOne.getString("schema_name");
+      // Check that schema name match the original input
+      // And check special case - schema with special name in quotes
+      boolean isSchemaNameMatch =
+          compiledSchemaPattern != null
+              && (compiledSchemaPattern.matcher(schemaName).matches()
+                  || (schemaName.startsWith("\"")
+                      && schemaName.endsWith("\"")
+                      && compiledSchemaPattern
+                          .matcher(schemaName)
+                          .region(1, schemaName.length() - 1)
+                          .matches()));
+
       // Check that procedure name and schema name match the original input in case wildcards have
       // been used.
       // Procedure name column check must occur later when columns are parsed.
       if ((compiledProcedurePattern != null
               && !compiledProcedurePattern.matcher(procedureNameNoArgs).matches())
-          || (compiledSchemaPattern != null
-              && !compiledSchemaPattern.matcher(schemaName).matches())) {
+          || !isSchemaNameMatch) {
```

Comment:
You need to do (compiledSchemaPattern != null
                            && !isSchemaNameMatch), so it cover the case when compiledSchemaPattern == NULL.

Reply:
Ok, thanks, added. Sorry for the late response.